File: src/main/java/no/nav/syfo/mappers/NaermesteLederMapper.java
package no.nav.syfo.mappers;

import no.nav.syfo.controller.domain.NaermesteLeder;
import no.nav.tjeneste.virksomhet.sykefravaersoppfoelging.v1.informasjon.WSNaermesteLeder;

import static org.apache.commons.lang3.text.WordUtils.capitalize;
public class NaermesteLederMapper {

    public static NaermesteLeder tilNaermesteLeder(WSNaermesteLeder response, String organisasjon) {
        return new NaermesteLeder()
                .withNavn(capitalize(response.getNavn()))
                .withEpost(response.getEpost())
                .withTlf(response.getMobil())
                .withFomDato(response.getNaermesteLederStatus().getAktivFom())
                .withOrgnummer(response.getOrgnummer())
                .withOrganisasjonsnavn(organisasjon)
                .withAktivTom(response.getNaermesteLederStatus().getAktivTom())
                .withArbeidsgiverForskuttererLoenn(response.isArbeidsgiverForskuttererLoenn())
                .withErOppgitt(true);
    }
}

File: src/main/kotlin/no/nav/syfo/mappers/NaermesteLederMapper.kt
package no.nav.syfo.mappers

import no.nav.syfo.controller.domain.NaermesteLeder
import no.nav.tjeneste.virksomhet.sykefravaersoppfoelging.v1.informasjon.WSNaermesteLeder
import org.apache.commons.lang3.text.WordUtils

fun tilNaermesteLeder(response: WSNaermesteLeder, organisasjon: String): NaermesteLeder {
    return NaermesteLeder()
        .withNavn(WordUtils.capitalize(response.navn))
        .withEpost(response.epost)
        .withTlf(response.mobil)
        .withFomDato(response.naermesteLederStatus.aktivFom)
        .withOrgnummer(response.orgnummer)
        .withOrganisasjonsnavn(organisasjon)
        .withAktivTom(response.naermesteLederStatus.aktivTom)
        .withArbeidsgiverForskuttererLoenn(response.isArbeidsgiverForskuttererLoenn)
        .withErOppgitt(true)
}
